ui/apps/platform/src/types/widgetConfig.ts

```diff
@@ -19,13 +19,13 @@ type WidgetConfigMap = Partial<Record<RouteId, WidgetConfig>>;
 export type RouteId = string;
 
 // Configuration read for a single instance of a widget.
-export type WidgetConfig = Partial<Record<string, ConfigOptionValue>>;
+export type WidgetConfig = Readonly<Partial<Record<string, ConfigOptionValue>>>;
```

Comment:
These structures being mutable has already caused some annoying bugs in development, so I've updated them to be `readonly`.